lib/charms/layer/container_runtime_common.py

```diff
@@ -0,0 +1,51 @@
+from subprocess import check_call, CalledProcessError, DEVNULL
+
+from ctl.containerd import ContainerdCtl
+from ctl.docker import DockerCtl
+
+
+def _detectContainerRuntime():
+    """
+    Attempt to detect the
+    container runtime.
+
+    :return: String name
+    """
+    try:
+        check_call(['ctr'], stdout=DEVNULL, stderr=DEVNULL)
```

Comment:
Is this safe? We expect to have ctr in the path? We could check if the containerd or docker service is present in systemctl.

Reply:
This is safe as it will always be in path if installed correctly.  However, I agree that it's nasty and would rather do it a better way.  Is there a charm helper to check this, or do we literally have to do an `os.path.isfile(path_to_service)`?